### DISCUSSION TITLE
Blender plugin for scene composition and building with complex meshes

### DISCUSSION THREAD

**Discussion by Balint-H** [google-deepmind/mujoco#861] [Show and tell]
Designing and aligning the elements of MJCF often involves a lot of trial and error, and makes precise alignment of parts difficult. With the Unity plugin we can construct scenes in a visual way, but Unity still lacks precise alignment and mesh editing tools. 

I've written a small WIP script for Blender that can take hierarchies of meshes in a blender scene and export an MJCF. I wanted to move the origins of different mesh objects to their axis of rotation, and automatically save the relative positions of child bodies, joints and sites (represented by empty objects in Blender). The produced MJCF will still be incomplete, and you still need to adjust existing and add missing elements in your text editor of choice. However, it will also export and reference mesh assets automatically.

A tool like this has been missing from my workflow, so hopefully others will find it useful too. Please feel free to fork or send PRs with fixes, features or improvements. I currently use simple object parenting and empty objects to represent the articulation, I wonder if anyone with more experience in rigging with Blender would suggest using Blender armatures instead.

Here's the repo:
https://github.com/Balint-H/blender-mjcf

**Comment by kevinzakka** [comment]
Nice!
